typescript/api/services/BrandingLogoService.ts

```diff
@@ -0,0 +1,159 @@
+        errors.push('empty');
+      }
+      if (!this.allowedContentTypes.has(contentType)) {
+        errors.push('unsupported-type');
+      }
+      const max = this.getMaxBytes();
+      if (fileBuf.length > max) {
+        errors.push('too-large');
+      }
+      let outBuffer = fileBuf;
+      let finalCt = contentType;
+      if (errors.length === 0 && this.isSvg(fileBuf, contentType)) {
+        // Sanitize SVG using existing service
+        const svg = fileBuf.toString('utf8');
+        const result = await SvgSanitizerService.sanitize(svg);
+        if (!result.safe) {
+          errors.push(...result.errors.map(e => 'svg-' + e));
+        }
+        warnings.push(...result.warnings.map(w => 'svg-' + w));
+        outBuffer = Buffer.from(result.sanitized, 'utf8');
+        finalCt = 'image/svg+xml';
+      }
+      if (errors.length) {
+        return { ok: false, errors, warnings };
+      }
+      const sha256 = crypto.createHash('sha256').update(outBuffer).digest('hex');
+      return { ok: true, sha256, sanitizedBuffer: outBuffer, warnings, finalContentType: finalCt };
+    }
+
+  /** Write logo to storage (GridFS) and update BrandingConfig.logo metadata */
+    async putLogo(opts: { branding: string; portal: string; fileBuffer: Buffer; contentType: string; actor?: any; }): Promise<{ hash: string; gridFsId: string; contentType: string; updatedAt: string; }> {
+      const brand = await BrandingConfig.findOne({ name: opts.branding });
+      if (!brand) throw new Error('branding-not-found');
+      const { ok, sha256, sanitizedBuffer, errors, finalContentType } = await this.sanitizeAndValidate(opts.fileBuffer, opts.contentType);
+      if (!ok) throw new Error('logo-invalid: ' + errors.join(','));
+      // Generate a deterministic filename-like id for traceability across storage layers
+      const ext = finalContentType === 'image/png' ? 'png' : finalContentType === 'image/jpeg' ? 'jpg' : 'svg';
+      const gridFsId = `${opts.branding}/${opts.portal}/images/logo.${ext}`;
+
+      // Persist to GridFS (if available). Keep in-memory cache for immediate serving.
+      try {
+        const bucket = this.getBucket();
+        if (bucket) {
+          // Delete any previous versions with same filename
+          const existing = await bucket.find({ filename: gridFsId }).toArray();
+          for (const f of existing) {
+            try { await bucket.delete(f._id); } catch(_e) { /* noop */ }
+          }
+          await new Promise<void>((resolve, reject) => {
+            const up = bucket.openUploadStream(gridFsId, { metadata: { contentType: finalContentType } });
+            up.on('error', reject);
+            up.on('finish', () => resolve());
+            up.end(sanitizedBuffer);
+          });
+        }
+      } catch(e) {
+        // Non-fatal: we still update config and serve from memory; log verbose only
+        sails.log && sails.log.warn && sails.log.warn('BrandingLogoService.putLogo GridFS write failed:', e);
```

Comment:
Why would there be no sails.log.warn?
Also, the comment says log verbose, but the log is done as a `warn`?